src/compositor/lipstickcompositor.cpp

```diff
@@ -700,9 +702,12 @@ void LipstickCompositor::setUpdatesEnabled(bool enabled)
                 m_onUpdatesDisabledUnfocusedWindowId = topmostWindow->windowId();
                 clearKeyboardFocus();
             }
-            hide();
-            if (QWindow::handle()) {
-                QGuiApplication::platformNativeInterface()->nativeResourceForIntegration("DisplayOff");
+            if (calledFromDBus()) {
+                setDelayedReply(true);
+                m_displayOffReply = message().createReply();
+                startTimer(1000);
```

Comment:
Yes, mce will wait until it gets the reply / lipstick drops from system bus.

The extra delays in on->off->on round trip might be annoying during calls / other places where display is turned off automatically. But IMO it would be better to get things working reliably asap - we can hopefully optimize this later on.